beetsplug/mpdupdate.py

```diff
@@ -112,18 +75,44 @@ def __init__(self):
             if self.config[key].exists():
                 config['mpd'][key] = self.config[key].get()
 
+        self.register_listener('database_change', self.db_change)
+
+    def db_change(self, lib):
+        self.register_listener('cli_exit', self.update)
```

Comment:
Interesting. I think this is the first time we've registered a listener anywhere other than the plugin setup sequence. If we ever move to automatic registration (with decorators or a naming convention), it may end up being easier to reason about the other style—as opposed to conditionally modifying the listeners, which could conflict with stuff like caching of the listener lists.